### PR TITLE
chore: better error messages for assertions

### PR DESCRIPTION
**Pull request by cprice404**

### CODE REVIEW

File: tests/Integration/Momento.Sdk.Tests/SimpleCacheControlTest.cs [hunk 1]
```diff
@@ -74,15 +74,15 @@ public async Task ListCachesAsync_OneCache_HappyPath()
 
         // Test cache exists
         ListCachesResponse result = await client.ListCachesAsync();
-        Assert.True(result is ListCachesResponse.Success);
+        Assert.True(result is ListCachesResponse.Success, $"Unexpected response: {result}");
```
Comment (poppoerika) [inline tests/Integration/Momento.Sdk.Tests/SimpleCacheControlTest.cs:77]: Do we need `Unexpected response` on Success?

File: tests/Integration/Momento.Sdk.Tests/SimpleCacheControlTest.cs [hunk 2]
```diff
@@ -74,15 +74,15 @@ public async Task ListCachesAsync_OneCache_HappyPath()
...
         var successResult = (ListCachesResponse.Success)result;
         List<CacheInfo> caches = successResult.Caches;
         Assert.Contains(new CacheInfo(cacheName), caches);
 
         // Test deleting cache
         await client.DeleteCacheAsync(cacheName);
         result = await client.ListCachesAsync();
-        Assert.True(result is ListCachesResponse.Success);
+        Assert.True(result is ListCachesResponse.Success, $"Unexpected response: {result}");
```
Comment (poppoerika) [inline tests/Integration/Momento.Sdk.Tests/SimpleCacheControlTest.cs:85]: https://github.com/momentohq/client-sdk-dotnet/pull/313/files#r1000982546

File: tests/Integration/Momento.Sdk.Tests/SimpleCacheControlTest.cs [hunk 3]
```diff
@@ -107,7 +107,7 @@ public async Task ListCachesAsync_Iteration_HappyPath()
         ListCachesResponse result = await client.ListCachesAsync();
         while (true)
         {
-            Assert.True(result is ListCachesResponse.Success);
+            Assert.True(result is ListCachesResponse.Success, $"Unexpected response: {result}");
```
Comment (poppoerika) [inline tests/Integration/Momento.Sdk.Tests/SimpleCacheControlTest.cs:110]: Same comment as https://github.com/momentohq/client-sdk-dotnet/pull/313/files#r1000982546

File: tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs [hunk 4]
```diff
@@ -56,7 +56,7 @@ public async Task SetAsync_KeyIsByteArrayValueIsByteArray_HappyPath()
         byte[] value = Utils.NewGuidByteArray();
         await client.SetAsync(cacheName, key, value);
         CacheGetResponse response = await client.GetAsync(cacheName, key);
-        Assert.True(response is CacheGetResponse.Hit);
+        Assert.True(response is CacheGetResponse.Hit, $"Unexpected response: {response}");
```
Comment (poppoerika) [inline tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs:59]: Same as https://github.com/momentohq/client-sdk-dotnet/pull/313/files#r1000982546

File: tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs [hunk 5]
```diff
@@ -65,7 +65,7 @@ public async Task SetAsync_KeyIsByteArrayValueIsByteArray_HappyPath()
         value = Utils.NewGuidByteArray();
         await client.SetAsync(cacheName, key, value, ttl: TimeSpan.FromSeconds(15));
         response = await client.GetAsync(cacheName, key);
-        Assert.True(response is CacheGetResponse.Hit);
+        Assert.True(response is CacheGetResponse.Hit, $"Unexpected response: {response}");
```
Comment (poppoerika) [inline tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs:68]: Same as https://github.com/momentohq/client-sdk-dotnet/pull/313/files#r1000982546

File: tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs [hunk 6]
```diff
@@ -115,10 +115,10 @@ public async Task SetAsync_KeyIsStringValueIsString_HappyPath()
         string key = Utils.NewGuidString();
         string value = Utils.NewGuidString();
         var setResponse = await client.SetAsync(cacheName, key, value);
-        Assert.True(setResponse is CacheSetResponse.Success);
+        Assert.True(setResponse is CacheSetResponse.Success, $"Unexpected response: {setResponse}");
```
Comment (poppoerika) [inline tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs:118]: Same as https://github.com/momentohq/client-sdk-dotnet/pull/313/files#r1000982546

File: tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs [hunk 7]
```diff
@@ -115,10 +115,10 @@ public async Task SetAsync_KeyIsStringValueIsString_HappyPath()
...
 
         CacheGetResponse response = await client.GetAsync(cacheName, key);
-        Assert.True(response is CacheGetResponse.Hit);
+        Assert.True(response is CacheGetResponse.Hit, $"Unexpected response: {response}");
```
Comment (poppoerika) [inline tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs:121]: Same as https://github.com/momentohq/client-sdk-dotnet/pull/313/files#r1000982546

File: tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs [hunk 8]
```diff
@@ -127,10 +127,10 @@ public async Task SetAsync_KeyIsStringValueIsString_HappyPath()
         key = Utils.NewGuidString();
         value = Utils.NewGuidString();
         setResponse = await client.SetAsync(cacheName, key, value, ttl: TimeSpan.FromSeconds(15));
-        Assert.True(setResponse is CacheSetResponse.Success);
+        Assert.True(setResponse is CacheSetResponse.Success, $"Unexpected response: {setResponse}");
```
Comment (poppoerika) [inline tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs:130]: Same as https://github.com/momentohq/client-sdk-dotnet/pull/313/files#r1000982546

File: tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs [hunk 9]
```diff
@@ -127,10 +127,10 @@ public async Task SetAsync_KeyIsStringValueIsString_HappyPath()
...
 
         response = await client.GetAsync(cacheName, key);
-        Assert.True(response is CacheGetResponse.Hit);
+        Assert.True(response is CacheGetResponse.Hit, $"Unexpected response: {response}");
```
Comment (poppoerika) [inline tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs:133]: Same as https://github.com/momentohq/client-sdk-dotnet/pull/313/files#r1000982546

File: tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs [hunk 10]
```diff
@@ -191,7 +191,7 @@ public async Task SetAsync_KeyIsStringValueIsByteArray_HappyPath()
         key = Utils.NewGuidString();
         value = Utils.NewGuidByteArray();
         setResponse = await client.SetAsync(cacheName, key, value, ttl: TimeSpan.FromSeconds(15));
-        Assert.True(setResponse is CacheSetResponse.Success);
+        Assert.True(setResponse is CacheSetResponse.Success, $"Unexpected response: {setResponse}");
```
Comment (poppoerika) [inline tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs:194]: Same as https://github.com/momentohq/client-sdk-dotnet/pull/313/files#r1000982546

File: tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs [hunk 11]
```diff
@@ -205,10 +205,10 @@ public async Task GetAsync_ExpiredTtl_HappyPath()
         string key = Utils.NewGuidString();
         string value = Utils.NewGuidString();
         var setResponse = await client.SetAsync(cacheName, key, value, TimeSpan.FromSeconds(1));
-        Assert.True(setResponse is CacheSetResponse.Success);
+        Assert.True(setResponse is CacheSetResponse.Success, $"Unexpected response: {setResponse}");
```
Comment (poppoerika) [inline tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs:208]: Same as https://github.com/momentohq/client-sdk-dotnet/pull/313/files#r1000982546

File: tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs [hunk 12]
```diff
@@ -205,10 +205,10 @@ public async Task GetAsync_ExpiredTtl_HappyPath()
...
         await Task.Delay(3000);
         CacheGetResponse result = await client.GetAsync(cacheName, key);
-        Assert.True(result is CacheGetResponse.Miss);
+        Assert.True(result is CacheGetResponse.Miss, $"Unexpected response: {result}");
```
Comment (poppoerika) [inline tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs:211]: Same as https://github.com/momentohq/client-sdk-dotnet/pull/313/files#r1000982546

File: tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs [hunk 13]
```diff
@@ -228,17 +228,17 @@ public async Task DeleteAsync_KeyIsByteArray_HappyPath()
         byte[] key = new byte[] { 0x01, 0x02, 0x03, 0x04 };
         byte[] value = new byte[] { 0x05, 0x06, 0x07, 0x08 };
         var setResponse = await client.SetAsync(cacheName, key, value, ttl: TimeSpan.FromMinutes(1));
-        Assert.True(setResponse is CacheSetResponse.Success);
+        Assert.True(setResponse is CacheSetResponse.Success, $"Unexpected response: {setResponse}");
```
Comment (poppoerika) [inline tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs:231]: Same as https://github.com/momentohq/client-sdk-dotnet/pull/313/files#r1000982546

File: tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs [hunk 14]
```diff
@@ -228,17 +228,17 @@ public async Task DeleteAsync_KeyIsByteArray_HappyPath()
...
         CacheGetResponse getResponse = await client.GetAsync(cacheName, key);
-        Assert.True(getResponse is CacheGetResponse.Hit);
+        Assert.True(getResponse is CacheGetResponse.Hit, $"Unexpected response: {getResponse}");
```
Comment (poppoerika) [inline tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs:233]: Same as https://github.com/momentohq/client-sdk-dotnet/pull/313/files#r1000982546

File: tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs [hunk 15]
```diff
@@ -228,17 +228,17 @@ public async Task DeleteAsync_KeyIsByteArray_HappyPath()
...
 
         // Delete
         var deleteResponse = await client.DeleteAsync(cacheName, key);
-        Assert.True(deleteResponse is CacheDeleteResponse.Success);
+        Assert.True(deleteResponse is CacheDeleteResponse.Success, $"Unexpected response: {deleteResponse}");
```
Comment (poppoerika) [inline tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs:237]: Same as https://github.com/momentohq/client-sdk-dotnet/pull/313/files#r1000982546

File: tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs [hunk 16]
```diff
@@ -228,17 +228,17 @@ public async Task DeleteAsync_KeyIsByteArray_HappyPath()
...
 
         // Check deleted
         getResponse = await client.GetAsync(cacheName, key);
-        Assert.True(getResponse is CacheGetResponse.Miss);
+        Assert.True(getResponse is CacheGetResponse.Miss, $"Unexpected response: {getResponse}");
```
Comment (poppoerika) [inline tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs:241]: Same as https://github.com/momentohq/client-sdk-dotnet/pull/313/files#r1000982546

File: tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs [hunk 17]
```diff
@@ -258,15 +258,15 @@ public async Task DeleteAsync_KeyIsString_HappyPath()
         string key = Utils.NewGuidString();
         string value = Utils.NewGuidString();
         var setResponse = await client.SetAsync(cacheName, key, value, ttl: TimeSpan.FromMinutes(1));
-        Assert.True(setResponse is CacheSetResponse.Success);
+        Assert.True(setResponse is CacheSetResponse.Success, $"Unexpected response: {setResponse}");
```
Comment (poppoerika) [inline tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs:261]: Same as https://github.com/momentohq/client-sdk-dotnet/pull/313/files#r1000982546

File: tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs [hunk 18]
```diff
@@ -258,15 +258,15 @@ public async Task DeleteAsync_KeyIsString_HappyPath()
...
         CacheGetResponse getResponse = await client.GetAsync(cacheName, key);
-        Assert.True(getResponse is CacheGetResponse.Hit);
+        Assert.True(getResponse is CacheGetResponse.Hit, $"Unexpected response: {getResponse}");
```
Comment (poppoerika) [inline tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs:263]: Same as https://github.com/momentohq/client-sdk-dotnet/pull/313/files#r1000982546

File: tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs [hunk 19]
```diff
@@ -258,15 +258,15 @@ public async Task DeleteAsync_KeyIsString_HappyPath()
...
 
         // Delete
         await client.DeleteAsync(cacheName, key);
 
         // Check deleted
         getResponse = await client.GetAsync(cacheName, key);
-        Assert.True(getResponse is CacheGetResponse.Miss);
+        Assert.True(getResponse is CacheGetResponse.Miss, $"Unexpected response: {getResponse}");
```
Comment (poppoerika) [inline tests/Integration/Momento.Sdk.Tests/SimpleCacheDataTest.cs:270]: Same as https://github.com/momentohq/client-sdk-dotnet/pull/313/files#r1000982546

**Comment by poppoerika** [comment]
I just left a lot of comments but they are basically asking: do we need `Unexpected response` message on `Success`, `Hit`, or `Miss`?

File: tests/Integration/Momento.Sdk.Tests/SimpleCacheControlTest.cs (hunk 1 above)
Comment (pgautier404) [reply tests/Integration/Momento.Sdk.Tests/SimpleCacheControlTest.cs:77]: @poppoerika These messages are printed when the assertion is not satisfied, so the result being printed in the message is **not** a success.

**Review by pgautier404 — APPROVED** [review]

File: tests/Integration/Momento.Sdk.Tests/SimpleCacheControlTest.cs (hunk 1 above)
Comment (poppoerika) [reply tests/Integration/Momento.Sdk.Tests/SimpleCacheControlTest.cs:77]: 👍

**Pull request merged**